spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaSettings.java

```diff
@@ -417,6 +427,220 @@ public void setFormat(@Nullable String format) {
         }
     }
 
+    /**
+     * Configuration for a Athenz provider.
+     * If this is set, you can use `@RequiresAthenzRole` annotation to secure your service methods.
+     */
+    @UnstableApi
+    public static class AthenzConfig {
```

Comment:
Question) Is it possible that the clas `AthenzConfig` lives separately from `ArmeriaSettings`, with only the prefix defined as `armeria.athenz`?

e.g.
```java
@ConfigurationProperties(prefix = "armeria.athenz")
class AthenzProperties {
```

Reply:
It is possible if `AthenzConfig` is registered to `@EnableConfigurationProperties()`.
Do you prefer moving `AthenzConfig` to the top-level class and injecting it into `AbstractArmeriaAutoConfiguration` independently?
https://github.com/line/armeria/blob/57295d3ed82748dc6949f825910705a6c17d098f/spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaAutoConfiguration.java#L35